tests/components/decora_wifi/test_common.py

```diff
@@ -0,0 +1,202 @@
+"""Test the decora_wifi Common components."""
+
+from unittest.mock import patch
+
+from homeassistant.components.decora_wifi.common import (
+    CommFailed,
+    DecoraWifiEntity,
+    DecoraWifiPlatform,
+    LoginFailed,
+)
+from homeassistant.helpers import device_registry
+
+from tests.components.decora_wifi.common import (
+    SWITCH_NAME,
+    FakeDecoraWiFiIotSwitch,
+    FakeDecoraWiFiResidence,
+    FakeDecoraWiFiResidentialAccount,
+    FakeDecoraWiFiSession,
+)
+
+USERNAME = "username@home-assisant.com"
+PASSWORD = "test-password"
+INCORRECT_PASSWORD = "incorrect-password"
+MODEL = "DW4SF"
+
```

Comment:
Tests should not interact with the integration details.

https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations

Setup the integration, call services to interact with the integration (patching the PyPI package code as needed) and assert the entity state via the core state machine.

Instead create a `test_light.py` and remove these tests.

Reply:
I think I made good progress on this. Not sure what happened to the workflows. GitHub is acting funny today.